platform/ios/src/MGLMapView.mm

```diff
@@ -4466,13 +4466,6 @@ - (void)notifyMapChange:(mbgl::MapChange)change
         case mbgl::MapChangeRegionWillChange:
         case mbgl::MapChangeRegionWillChangeAnimated:
         {
-            if ( ! _userLocationAnnotationIsSelected
-                || self.userTrackingMode == MGLUserTrackingModeNone
-                || self.userTrackingState != MGLUserTrackingStateChanged)
-            {
-                [self deselectAnnotation:self.selectedAnnotation animated:NO];
```

Comment:
The annotation still needs to be deselected if it lies outside the viewport. See `-selectAnnotation:animated:`.